src/modal/modal.ts

```diff
@@ -366,8 +393,7 @@ export class CaptureEyeModal extends LitElement {
     // 0: User-customized link (not provided by us)
     // 1: Default engagement link
     // 2, 3, ...: Future rotating engagement links
-    const subid =
-      this.engagementLink === Constant.url.defaultEngagementLink ? '1' : '0';
+    const subid = this._engagementLink ? '0' : '1';
```

Comment:
The default value is defaultEngagementLink, so it's probably best to keep the original logic. Also, when the user customizes the value to match our default link, the value should be set to '1'.

Reply:
@olgahaha In this update I changed the protected states like `this._engagementLink` to override-only (defaults to empty string. If it's empty then when rendering use default value. Only if not empty then use the custom value for overriding. So the new default value of `this._engagementLink` is empty, but I can see your point we should also treat when `eng-link` attribute is set to the same value as default link, we should also set it to 1.